### DISCUSSION TITLE
Tuya TS0601 Human Presence Detector

### DISCUSSION THREAD

**Discussion by losip** [Koenkk/zigbee2mqtt#10254] [Devices]
A few weeks ago I was looking for a human presence detector and, on the supported devices list, I found a new Tuya TS0601 device with a sophisticated radar and "fall down" detector so I sent off for one from here: https://www.aliexpress.com/item/1005003525430429.html?spm=a2g0s.9042311.0.0.3a504c4dVYRCaf.  This was not the one that is listed at https://www.zigbee2mqtt.io/devices/TS0601_motion_sensor.html#tuya-ts0601-motion-sensor.

It has now arrived and I paired it with Z2M to be greeted by a continuous stream of messages containg " Received message from unsupported device with Zigbee model 'TS0601' and manufacturer name '_TZE200_lu01t0zl'". So back to the supported devices list, which has a new search function and I can't find it.

Before I enter the dark waters of adding support for a new Tuya device, I thought I'd ask if anyone knows anything about this device (which was rather expensive). Not being a programmer, I don't want to reinvent the wheel if it already exists.

Thanks

**Comment by losip** [comment]
With further research I have found that there is a converter for a similar radar device (manufacturerName: '_TZE200_vrfecyku') while mine is manufacturerName: '_TZE200_lu01t0zl'. Is there any way of spoofing a Tuya device to use the known name?

**Reply by bladehstream** [reply]
Looks like this has now been merged:
https://github.com/Koenkk/zigbee-herdsman-converters/blob/master/devices/tuya.js#L3169

Can you confirm it's working fine?  Also, could you update the link?

**Comment by joohann** [comment]
New manufacturerName: _TZE204_ztc6ggyl

**Reply by Koenkk** [reply]
Added! Will be supported in tomorrow's release.

**Comment by brunialti** [comment]
After few tests I found ts0601_smart_human_presence_sensor too chatty. At least 2 msgs per second looking at mqttexplorer. Even worse after a while (say 1vhour or less) ,It stuck to presence=true. May be It Is engulfed by too many messages.

**Reply by wolph** [reply]
I just received mine and I have the same issue. It's spamming messages throughout my zigbee network like crazy and changing the settings doesn't make a difference.

To illustrate:
```
Zigbee2MQTT:info  2022-12-13 16:24:35: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945075719,"linkquality":120,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.8}'
Zigbee2MQTT:info  2022-12-13 16:24:36: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945076725,"linkquality":120,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.98}'
Zigbee2MQTT:info  2022-12-13 16:24:37: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945077721,"linkquality":123,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":3.02}'
Zigbee2MQTT:info  2022-12-13 16:24:38: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945078720,"linkquality":123,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":3}'
Zigbee2MQTT:info  2022-12-13 16:24:39: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945079720,"linkquality":123,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.94}'
Zigbee2MQTT:info  2022-12-13 16:24:40: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945080718,"linkquality":120,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.89}'
Zigbee2MQTT:info  2022-12-13 16:24:42: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945082719,"linkquality":127,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.84}'
Zigbee2MQTT:info  2022-12-13 16:24:43: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945083718,"linkquality":127,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.89}'
Zigbee2MQTT:info  2022-12-13 16:24:44: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945084719,"linkquality":127,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":3.03}'
Zigbee2MQTT:info  2022-12-13 16:24:45: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945085718,"linkquality":123,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":3.44}'
Zigbee2MQTT:info  2022-12-13 16:24:46: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945086721,"linkquality":127,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":3.14}'
Zigbee2MQTT:info  2022-12-13 16:24:47: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945087719,"linkquality":120,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.89}'
```

**Reply by gaussey** [reply]
> I just received mine and I have the same issue. It's spamming messages throughout my zigbee network like crazy and changing the settings doesn't make a difference.
> 
> To illustrate:
> 
> ```
> Zigbee2MQTT:info  2022-12-13 16:24:35: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945075719,"linkquality":120,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.8}'
> Zigbee2MQTT:info  2022-12-13 16:24:36: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945076725,"linkquality":120,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.98}'
> Zigbee2MQTT:info  2022-12-13 16:24:37: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945077721,"linkquality":123,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":3.02}'
> Zigbee2MQTT:info  2022-12-13 16:24:38: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945078720,"linkquality":123,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":3}'
> Zigbee2MQTT:info  2022-12-13 16:24:39: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945079720,"linkquality":123,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.94}'
> Zigbee2MQTT:info  2022-12-13 16:24:40: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945080718,"linkquality":120,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.89}'
> Zigbee2MQTT:info  2022-12-13 16:24:42: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945082719,"linkquality":127,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.84}'
> Zigbee2MQTT:info  2022-12-13 16:24:43: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945083718,"linkquality":127,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.89}'
> Zigbee2MQTT:info  2022-12-13 16:24:44: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945084719,"linkquality":127,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":3.03}'
> Zigbee2MQTT:info  2022-12-13 16:24:45: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945085718,"linkquality":123,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":3.44}'
> Zigbee2MQTT:info  2022-12-13 16:24:46: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945086721,"linkquality":127,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":3.14}'
> Zigbee2MQTT:info  2022-12-13 16:24:47: MQTT publish: topic 'zigbee2mqtt/0x70ac08fffeb77205', payload '{"cli":" ","detection_delay":5,"fading_time":120,"illuminance_lux":42,"last_seen":1670945087719,"linkquality":120,"maximum_range":9.5,"minimum_range":0.6,"presence":true,"radar_sensitivity":9,"self_test":"check_success","target_distance":2.89}'
> ```

Mine just came too and holy crap, RIP logs.

**Reply by menzbenz-jeedom** [reply]
Hello everybody, i whish you happy new year ! 

Any solution for this ? i have the same problem .
Thank you :)

**Reply by wolph** [reply]
> Any solution for this ? i have the same problem . Thank you :)

2 options.

1. buy a different sensor
2. replace the zigbee chip with an esp32 or esp8266

There's also the chance that Tuya will release a new firmware somewhere in the future that could fix this. But I'm not holding my breath on that one

**Reply by menzbenz-jeedom** [reply]
Hi @wolph , Thank you for helping. 
I think is not possible for me to change the chip as i have a generic box (jeedom zigbee atlas). I think that il will buy a different sensor. Have you any idea about zigbee sensor that works on AC (100V-220V) that is not very expensive please ? like this one but not Tuya please : https://www.zigbee2mqtt.io/devices/TS0601_smart_human_presence_sensor.html

Thank you

**Reply by wolph** [reply]
I've only tried these things and the Aqara FP1. The FP1 works great but it's a lot less pretty/convenient solution so I was really rooting for these things.

For now I plan on replacing the zigbee chips with an ESP32 like this: https://blakadder.com/zy-m100/#tasmota

It's rather inconvenient but I've got a few of these sensors and the sensors themselves seem to work just fine.

**Reply by menzbenz-jeedom** [reply]
Hi @wolph,
Thank you for all information .
I agree with you , aqara fp1 is less pretty then Tuya.  
I could not change the chip as i already bought this universal hub https://www.domadoo.fr/fr/box-domotique/5878-jeedom-controleur-domotique-jeedom-atlas-zigbee.html . It was very expressive for me, i could not buy more component as i want to keep the less of bulk. 
In addition, i found this device which is compatible with Z2M https://www.zigbee2mqtt.io/devices/TS0601_motion_sensor.html

And there is no "target distance " according to the description .
I found it also in Aliexpress :
https://a.aliexpress.com/_EJ8kxyx
Do you think that it will be a good idea or it will be throw money away ?

Thank you :)

**Reply by wolph** [reply]
I've only tried the FP1 and the Tuya so far.

From what I've seen the Tuya works fine, but it's sending out an insane amount of updates and since I bought 4 of these and have a pretty large Zigbee network (60 devices not including these sensors) I've got a feeling that they'll bog down my Zigbee network if they're all sending 10 updates per second ;)

In your case I would suggest trying the device and see if it works for you. Perhaps there will be a firmware update that fixes this somewhere in the future.

**Reply by menzbenz-jeedom** [reply]
Hi @wolph,

I see! I will try this sensor and keep you informed! 
I already bought one . 
Have a nice  day.  

Thank you

**Reply by menzbenz-jeedom** [reply]
Hi again @wolph,

I received and tried the new motion sensor "https://www.zigbee2mqtt.io/devices/TS0601_motion_sensor.html"
It is much better then the https://www.zigbee2mqtt.io/devices/TS0601_smart_human_presence_sensor.html
I receive between 4 and 12 messages per minutes when presence detected . What do you think about ? is it lots of messages ?  as I have to by 7 sensors more .

Remark: the TS0601_motion_sensor is at least twice bigger than TS0601_smart_human_presence_sensor.
Thank you :)

**Reply by wolph** [reply]
@menzbenz-jeedom That seems like a much more reasonable amount of messages :)

It looks like a very nice sensor, but I'm not entirely sure if it's the same class of device.

A motion sensor is not the same as a presence sensor. A motion sensor detects motion, so if you stay still it will eventually think you have left the room.
A presence sensor is sensitive enough to detect if you're breathing even if you stay perfectly still.

**Reply by menzbenz-jeedom** [reply]
@wolph thank you for you quick reply! Yes I agree with you is not exatcly the same device class. The motion sensor supply the most important feature for me , presence(yes i think it is less smarter than the 'smart human sensor') and light level 'lux'. The smart human presence supply event the postion if the person she is standing or she is sleeping but unfortantly it crashe my smart home box . 
Im happy with this new sensor , it is better than nothing :D
Thank you @wolph , have a nice day

**Reply by menzbenz-jeedom** [reply]
![image](https://user-images.githubusercontent.com/122239803/213207116-fa2028e4-d5bf-465c-88ac-534e3952b025.png)

**Comment by unxplained** [comment]
Hi, I have a new manufacturer name for the same device "_TZE204_ijxvkhd0"

**Comment by lukas9956** [comment]
Got one few days ago and still can't get through it. Manufactuer name: "_TZE200_sgpeacqp"

**Reply by wiktorwar** [reply]
You can use this gist: https://gist.github.com/Koenkk/9295fc8afcc65f36027f9ab4d319ce64
copy it to `zigbee2mqtt/ext_converter.js` (new file) 
add this line to `configuration.yaml`:
```
external_converters:
  - ext_converter.js
```

And replace with your manufacturer name (`_TZE200_sgpeacqp` ) in the `ext_converter.js`

**Reply by wiktorwar** [reply]
Tried it myself and it doesn't work 😞 
@Koenkk do I need to switch to develop to use legacy converters? 
e.g. `legacy.fz.tuya_smart_human_presense_sensor`

**Reply by Koenkk** [reply]
No, I updated https://gist.github.com/Koenkk/9295fc8afcc65f36027f9ab4d319ce64

**Reply by wiktorwar** [reply]
here is a working converter (for `_TZE200_sgpeacqp `) : https://gist.github.com/wiktorwar/00ca0499e3e8ae4a4e604e2a744ba4cb

**Comment by renssies** [comment]
When researching my device I came across this discussion. Mine doesn't seem to be so chatty, it updates the light level a lot, but only a maximum of once per few seconds. 

However I have the perfect usecase for a mmWave sensor with built in relay, so I want to confirm if this is the same device we are talking about. 

The brand on the box is Wenzhi and type on the box is MTG235-ZB-RL/WZ-235-ZB-RL. It does say version 1.1

<details><summary>Label/Size Photo (click to expand)</summary>
<p>

![IMG_8434](https://github.com/user-attachments/assets/41c9bca6-d07e-419d-acd0-3cd6d0f40b46)

</p>
</details> 

<details><summary>Top photo (click to expand)</summary>
<p>

![IMG_8435](https://github.com/user-attachments/assets/310fff3a-ef7c-46f6-a607-e13606700e61)

</p>
</details> 

<details><summary>Zigbee 2 MQTT details (click to expand)</summary>
<p>

<img width="609" height="647" alt="Screenshot 2025-09-04 at 09 47 15" src="https://github.com/user-attachments/assets/f5ba84c0-b3ca-4c09-96b7-8d1334225f25" />

</p>
</details>
